packages/reactive-element/src/test/decorators/queryAssignedElements_test.ts

```diff
@@ -0,0 +1,195 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {queryAssignedElements} from '../../decorators/query-assigned-elements.js';
+import {
+  canTestReactiveElement,
+  generateElementName,
+  RenderingElement,
+  html,
+} from '../test-helpers.js';
+import {assert} from '@esm-bundle/chai';
+
+const flush =
+  window.ShadyDOM && window.ShadyDOM.flush ? window.ShadyDOM.flush : () => {};
+
+(canTestReactiveElement ? suite : suite.skip)('@queryAssignedElements', () => {
+  let container: HTMLElement;
+  let el: C;
+
+  class D extends RenderingElement {
+    @queryAssignedElements() defaultAssigned!: Element[];
+
+    @queryAssignedElements({slotName: 'footer', flatten: true})
+    footerAssigned!: Element[];
+
+    @queryAssignedElements({slotName: 'footer', flatten: false})
+    footerNotFlattenedSlot!: Element[];
+
+    @queryAssignedElements({
+      slotName: 'footer',
+      flatten: true,
+      selector: '.item',
+    })
+    footerAssignedFiltered!: Element[];
+
+    override render() {
+      return html`
+        <slot></slot>
+        <slot name="footer"></slot>
+      `;
+    }
+  }
+  customElements.define('assigned-elements-el', D);
+
+  class E extends RenderingElement {
+    @queryAssignedElements() defaultAssigned!: Element[];
+
+    @queryAssignedElements({slotName: 'header'}) headerAssigned!: Element[];
+
+    override render() {
+      return html`
+        <slot name="header"></slot>
+        <slot></slot>
+      `;
+    }
+  }
+  customElements.define('assigned-elements-el-2', E);
+
+  const defaultSymbol = Symbol('default');
+  const headerSymbol = Symbol('header');
+  class S extends RenderingElement {
+    @queryAssignedElements() [defaultSymbol]!: Element[];
```

Comment:
This is really just testing that `decorateProperty()` works with symbols. That should be tested once, not with every decorator. If you want to be extra sure that this works in case it's refactored, one symbol-keyed property among the rest in the existing classes should suffice.

Reply:
Great point. Left the default symbol case (for extra certainty) but removed the header symbol.

Reply:
I think you can remove this element and just make one of the fields in one of the other classes a symbol.

Reply:
Done!